test/integration/metrics_service_integration_test.cc

```diff
@@ -65,12 +65,12 @@ class MetricsServiceIntegrationTest : public HttpIntegrationTest,
     bool known_gauge_exists = false;
     for (::io::prometheus::client::MetricFamily metrics_family : envoy_metrics) {
       if (metrics_family.name() == "cluster.metrics_service.membership_change" &&
-          metrics_family.metric(0).has_counter()) {
+          metrics_family.type() == ::io::prometheus::client::MetricType::COUNTER) {
         known_counter_exists = true;
         EXPECT_EQ(1, metrics_family.metric(0).counter().value());
```

Comment:
I'd say it's also worth an ASSERT on has_timestamp_ms() outside of the if() to make sure that as new types are added folks don't forget to set that field.